src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/EDateEOMonth.txt

```diff
@@ -0,0 +1,566 @@
+
+// EDate basic tests
+
+>> EDate(Date(1981,1,1),1)
+Date(1981,2,1)
+
+>> EDate(Date(1981,1,15),1)
+Date(1981,2,15)
+
+>> EDate(Date(1981,1,27),1)
+Date(1981,2,27)
+
+>> EDate(Date(1981,1,28),1)
+Date(1981,2,28)
+
+>> EDate(Date(1981,1,29),1)
+Date(1981,2,28)
+
+>> EDate(Date(1981,1,30),1)
+Date(1981,2,28)
+
+>> EDate(Date(1981,1,31),1)
+Date(1981,2,28)
+
+>> EDate(Date(1981,2,1),1)
+Date(1981,3,1)
+
+>> EDate(Date(1981,2,28),1)
+Date(1981,3,28)
+
+>> EDate(Date(1981,2,1),0)
+Date(1981,2,1)
+
+>> EDate(Date(1981,2,28),0)
+Date(1981,2,28)
+
+>> EDate(Date(1981,3,30),1)
+Date(1981,4,30)
+
+>> EDate(Date(1981,3,1),0)
+Date(1981,3,1)
+
+>> EDate(Date(1981,3,30),0)
+Date(1981,3,30)
+
+>> EDate(Date(1981,12,1),1)
+Date(1982,1,1)
+
+>> EDate(Date(1981,12,15),1)
+Date(1982,1,15)
+
+>> EDate(Date(1981,12,30),1)
+Date(1982,1,30)
+
+>> EDate(Date(1981,12,31),1)
+Date(1982,1,31)
+
+>> EDate(Date(1981,12,1),2)
+Date(1982,2,1)
+
+>> EDate(Date(1981,12,15),2)
+Date(1982,2,15)
+
+>> EDate(Date(1981,12,30),2)
+Date(1982,2,28)
+
+>> EDate(Date(1981,12,31),2)
+Date(1982,2,28)
+
+// EOMonth basic tests
+
+>> EOMonth(Date(1981,1,1),1)
+Date(1981,2,28)
+
+>> EOMonth(Date(1981,1,15),1)
+Date(1981,2,28)
+
+>> EOMonth(Date(1981,1,28),1)
+Date(1981,2,28)
+
+>> EOMonth(Date(1981,1,29),1)
+Date(1981,2,28)
+
+>> EOMonth(Date(1981,1,30),1)
+Date(1981,2,28)
+
+>> EOMonth(Date(1981,1,31),1)
+Date(1981,2,28)
+
+>> EOMonth(Date(1981,2,1),1)
+Date(1981,3,31)
+
+>> EOMonth(Date(1981,2,28),1)
+Date(1981,3,31)
+
+>> EOMonth(Date(1981,2,1),0)
+Date(1981,2,28)
+
+>> EOMonth(Date(1981,2,28),0)
+Date(1981,2,28)
+
+>> EOMonth(Date(1981,3,30),1)
+Date(1981,4,30)
+
+>> EOMonth(Date(1981,3,1),0)
+Date(1981,3,31)
+
+>> EOMonth(Date(1981,3,30),0)
+Date(1981,3,31)
+
+>> EOMonth(Date(1981,12,1),1)
+Date(1982,1,31)
+
+>> EOMonth(Date(1981,12,15),1)
+Date(1982,1,31)
+
+>> EOMonth(Date(1981,12,30),1)
+Date(1982,1,31)
+
+>> EOMonth(Date(1981,12,31),1)
+Date(1982,1,31)
+
+>> EOMonth(Date(1981,12,1),2)
+Date(1982,2,28)
+
+>> EOMonth(Date(1981,12,15),2)
+Date(1982,2,28)
+
+>> EOMonth(Date(1981,12,30),2)
+Date(1982,2,28)
+
+>> EOMonth(Date(1981,12,31),2)
+Date(1982,2,28)
+
+// Leap year
+
+>> EDate(Date(2016,1,31),1)
+Date(2016,2,29)
+
+>> EOMonth(Date(2016,1,31),1)
+Date(2016,2,29)
+
+>> EDate(Date(2016,1,15),1)
+Date(2016,2,15)
+
+>> EOMonth(Date(2016,1,15),1)
+Date(2016,2,29)
+
+>> EDate(Date(2016,1,15),2)
+Date(2016,3,15)
+
+>> EOMonth(Date(2016,1,15),2)
+Date(2016,3,30)
+
+// Fractional months, should truncate (toward zero)
+
+>> EDate(Date(2012,1,31),1)
+Date(2016,2,29)
```

Comment:
This result (and many others after that) are incorrect.

Reply:
Done.